addons/common/functions/fnc_isEngineer.sqf

```diff
@@ -15,10 +17,11 @@
  * Public: Yes
  */
 
-params ["_unit"];
+params ["_unit", ["_engineerN", 1]];
 
 private _isEngineer = _unit getVariable ["ACE_isEngineer", _unit getUnitTrait "engineer"];
-//Handle ace_repair modules setting this to a number
-if (_isEngineer isEqualType 0) then {_isEngineer = _isEngineer > 0};
 
-_isEngineer
+// Convert boolean to number
+_isEngineer = [0, 1, 2] select _isEngineer;
+
+_isEngineer >= _engineerN
```

Comment:
Can you test, which is faster: Your current code or
```sqf
if (_isEngineer isEqualType 0) then {_isEngineer = _isEngineer > _engineerN};

_isEngineer
```
?

Reply:
I changed that to cover cases where `ACE_isEngineer` is `true` rather than a number

Reply:
Ah sorry, missed that.